renderdoc/driver/vulkan/vk_replay.h

```diff
@@ -727,4 +727,30 @@ class VulkanReplay : public IReplayDriver
   AMDRGPControl *m_RGP = NULL;
 
   VulkanAMDDrawCallback *m_pAMDDrawCallback = NULL;
+
+  std::vector<CounterResult> FetchCountersKHR(const std::vector<GPUCounter> &counters);
+
+  // Map a KHR counter to a Renderdoc one
+  struct KHRCounterMapping
+  {
+    KHRCounterMapping(CounterUnit _unit, CompType _type, uint32_t _byteWidth)
+      : unit(_unit)
+      , type(_type)
+      , byteWidth(_byteWidth) {}
+
+    CounterUnit unit;
+    CompType type;
+    uint32_t byteWidth;
+  };
+
+  std::map<VkPerformanceCounterUnitKHR, KHRCounterMapping> m_KHRCounterUnitsMap;
```

Comment:
Why is this stored as a dynamic map when the enum values and mapping is known at compile-time? This would be better as an static function in `vk_counters.cpp` that does switch/return.

Reply:
Done.